### ISSUE TITLE
XFCE (xfwm4) crashes on unstable

### ISSUE THREAD

**Issue by rgrunbla** [NixOS/nixpkgs#264547]
### Describe the bug

xfwm4 crashes on nixos-unstable for no apparent reason.

### Steps To Reproduce

1. Launch a VM reproducing the problem using `nix build github:rgrunbla/nixos-xfce-crash#vm && ./result`.
2. Login (keymap: us, password: `password`)
3. Wait for a bit, xfwm4 should crash. If you open a terminal, window decoration is missing, the toolbar should be absent…

Here is some screencap of what happens here :

https://github.com/NixOS/nixpkgs/assets/42433779/73cdebab-5ea4-44fa-a6f1-eb9607734884

The configuration of the VM is simple enough there should be no problem with it, but… it's possible there is an error in it. Here is the configuration file : https://github.com/rgrunbla/nixos-xfce-crash/blob/main/configuration.nix

Here is some backtrace I caught (but that was before I created this "minimal" example) : 
[backtrace.txt](https://github.com/NixOS/nixpkgs/files/13213868/backtrace.txt)

### Notify maintainers

@bobby285271 
@romildo 
@muscaln 

### Metadata

In the VM:

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.60, NixOS, 23.11 (Tapir), 23.11.20231026.63678e9`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.1`
 - nixpkgs: `not found`
```

**Comment by kirillrdy** [comment]
possibly fixed by https://github.com/NixOS/nixpkgs/pull/264230 ?

**Comment by rgrunbla** [comment]
> possibly fixed by #264230 ?

Possibly, I'm in the process of using that to check if that crashes. Thanks !

**Comment by rgrunbla** [comment]
Yeah, that fixes it, it seems ! Thanks @kirillrdy !
